src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -0,0 +1,502 @@
+
+                //TODO try catch
+                var convertedCompletionType = (IronPythonCompletionData.CompletionType)Enum.Parse(typeof(IronPythonCompletionData.CompletionType),
+                     Enum.GetName(typeof(ExternalCodeCompletionType), completion.Item4));
+
+                items.Add(new IronPythonCompletionData(completion.Item1, completion.Item2, completion.Item3, convertedCompletionType, this));
+
+            }
+            return items;
+
+
+
+        }
+
+        /// <summary>
+        /// List all of the members in a CLR type
+        /// </summary>
+        /// <param name="type">The type</param>
+        /// <param name="name">The name for the type</param>
+        /// <returns>A list of completion data for the type</returns>
+        protected List<IronPythonCompletionData> EnumerateMembers(Type type, string name)
+        {
+            var items = new List<IronPythonCompletionData>();
+            foreach (var completion in (providerImplementation as ILegacyPythonCompletionCore).EnumerateMembers(type, name))
+            {
+                //convert generalCompletiontype to legacy completionType
+
+                //TODO try catch
+                var convertedCompletionType = (IronPythonCompletionData.CompletionType)Enum.Parse(typeof(IronPythonCompletionData.CompletionType),
+                     Enum.GetName(typeof(ExternalCodeCompletionType), completion.Item4));
+
+                items.Add(new IronPythonCompletionData(completion.Item1, completion.Item2, completion.Item3, convertedCompletionType, this));
+
+            }
+            return items;
+
+
+
+        }
+
+        /// <summary>
+        /// Recursively lookup a member in a given namespace.
+        /// </summary>
+        /// <param name="name">A name for a type, possibly delimited by periods.</param>
+        /// <param name="n">The namespace</param>
+        /// <returns>The type as an object</returns>
+        public object LookupMember(string name, NamespaceTracker n)
+        {
+            return (this.providerImplementation as ILegacyPythonCompletionCore).LookupMember(name, n);
+        }
+
+        /// <summary>
+        /// Recursively lookup a variable in the _scope
+        /// </summary>
+        /// <param name="name">A name for a type, possibly delimited by periods.</param>
+        /// <returns>The type as an object</returns>
+        public object LookupMember(string name)
+        {
+            return (this.providerImplementation as ILegacyPythonCompletionCore).LookupMember(name);
+        }
+
+        /// <summary>
+        /// Generates completion data for the specified text. The text should be everything before
+        /// the dot character that triggered the completion. The text can contain the command line prompt
+        /// '>>>' as this will be ignored.
+        /// </summary>
+        [Obsolete("Please use additional GetDescription method as this one will be removed in Dynamo 3.0.")]
```

Comment:
Why tag specific methods from this class as obsolete when the entire class is going to be obsoleted?

Reply:
I did not add this, this obsoletion message was already here, but since I am moving this file back to this project everything appears like an "addition"

Reply:
I can remove these specific obsolete attributes if you think thats more clear and just keep the class level one?